File: android/src/main/java/br/com/brasizza/marcus/elgin/Printer.java
package br.com.brasizza.marcus.elgin;
import android.app.Activity;
import android.util.Base64;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Log;

import com.elgin.e1.Impressora.Termica;
import com.elgin.e1.Impressora.Utilidades.Inteiro;

import java.io.File;
import java.util.Map;

public class Printer {
    
    Activity mActivity;

    public Printer(Activity activity) {
        this.mActivity = activity;
        Termica.setActivity(this.mActivity);

    }

    public int printerInternalImpStart(Map map) {
        int typeImp =  (Integer) map.get("type");
        String modelImp =  (String) map.get("model");
        String connImp =  (String) map.get("connection");
        int paramImp =  (Integer) map.get("param");

        Log.d("elgin" ,"close connection");
        printerStop();
        Log.d("elgin" ,"printer closed");

        Log.d("elgin" ,"typeImp = "+ typeImp);
        Log.d("elgin" ,"modelImp = "+ modelImp);
        Log.d("elgin" ,"connImp = "+ connImp);
        Log.d("elgin" ,"paramImp = "+ paramImp);

        int result = Termica.AbreConexaoImpressora(typeImp, modelImp, connImp, paramImp);
        return result;
    }

  

    public void printerStop() {
        Termica.FechaConexaoImpressora();
    }

    public int avancaLinhas(int lines) {

        return Termica.AvancaPapel(lines);
    }

    public int cutPaper(int lines) {
        return Termica.Corte(lines);
    }

    public int imprimeTexto(Map map) {
        String text = (String) map.get("text");
        int align = (Integer) map.get("align");
        int font = (Integer) map.get("font");
        int fontSize = (Integer) map.get("fontSize");
        int result;
        result = Termica.ImpressaoTexto(text, align, font, fontSize);
        return result;
    }





    public int imprimeBarCode(Map map) {
        String text = (String) map.get("text");
        int barCodeType =  (Integer) map.get("barCodeType");
        int height = (Integer) map.get("height");
        int width = (Integer) map.get("width");
        int textPosition = (Integer) map.get("textPosition");
        int align = (Integer) map.get("align");
        Termica.DefinePosicao(align);
        return Termica.ImpressaoCodigoBarras(barCodeType, text, height, width, textPosition);
    }

    public int imprimeQR_CODE(Map map) {
        int size = (Integer) map.get("size");
        String text = (String) map.get("text");
        int align =  (Integer) map.get("align");
        int correctionLevel =  (Integer) map.get("correction");
        int result;
        int alignValue;
        Termica.DefinePosicao(align);
        result = Termica.ImpressaoQRCode(text, size, correctionLevel);
        return result;
    }


    public int printRaw(Map map) {
        byte[]  dataRaw =  (byte[]) map.get("data");
        int totalbytes = (Integer) map.get("bytes");
        Inteiro leu = new Inteiro(1);
        byte[] ler = new byte[1000];
        int ret =  Termica.DirectIO(dataRaw,totalbytes , ler, leu);
        return ret;

    }

    public int imprimeImagem(Map map) {
        String pathImage = (String) map.get("path");
        boolean isBase64 = (boolean) map.get("isBase64");

        int result;
        Bitmap bitmap;
        if(isBase64){
            byte[] decodedString = Base64.decode(pathImage, Base64.DEFAULT);
            bitmap = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
            
        }else{
            File mSaveBit = new File(pathImage); // Your image file
            String filePath = mSaveBit.getPath();
            bitmap = BitmapFactory.decodeFile(filePath);
        }

        result = Termica.ImprimeBitmap(bitmap);
        return result;
    }

    public int imprimeXMLNFCe(Map map) {
        String xmlNFCe = (String) map.get("xmlNFCe");
        System.out.println(xmlNFCe);
        int indexcsc = (int) map.get("indexcsc");
        String csc = (String) map.get("csc");
        int param = (int) map.get("param");
        return Termica.ImprimeXMLNFCe(xmlNFCe, indexcsc, csc, param);
    }

    public int imprimeXMLSAT(Map map) {
        String xml = (String) map.get("xmlSAT");
        int param = (int) map.get("param");
        return Termica.ImprimeXMLSAT(xml, param);
    }

    public int imprimeCupomTEF(Map map){
        String base64 = (String) map.get("base64");

        return Termica.ImprimeCupomTEF(base64);
    }

    public int statusGaveta() {
        return Termica.StatusImpressora(1);
    }

    public int statusSensorPapel() {
        return Termica.StatusImpressora(3);
    }

    public int statusEjetor() {
        return Termica.StatusImpressora(4);
    }

    public int abrirGavetaElgin() { return Termica.AbreGavetaElgin(); }

    public int abrirGaveta(Map map){
        int pin = (int) map.get("pin");
        int it = (int) map.get("it");
        int dp = (int) map.get("dp");
        return Termica.AbreGaveta(pin,it,dp);
    }


    public int sinalSonoro(Map map){
        int qtdSinal = (int) map.get("times");
        int st = (int) map.get("st");
        int ft = (int) map.get("ft");
        return Termica.SinalSonoro(qtdSinal,st,ft);
    }


    public String versaoImpressora(){
        return Termica.GetVersaoDLL();
    }


    public int InicializaImpressora(){ return  Termica.InicializaImpressora(); }

}

File: android/src/main/kotlin/br/com/brasizza/marcus/elgin/Printer.kt
package br.com.brasizza.marcus.elgin

import android.app.Activity
import android.util.Base64
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.util.Log
import com.elgin.e1.Impressora.Termica
import com.elgin.e1.Impressora.Utilidades.RetornoPOS
import java.io.File

class Printer(private val mActivity: Activity) {

    init {
        Termica.setActivity(this.mActivity)
    }

    fun printerInternalImpStart(map: Map<String, Any>): Int {
        val typeImp = map["type"] as Int
        val modelImp = map["model"] as String
        val connImp = map["connection"] as String
        val paramImp = map["param"] as Int

        Log.d("elgin", "close connection")
        printerStop()
        Log.d("elgin", "printer closed")

        Log.d("elgin", "typeImp = $typeImp")
        Log.d("elgin", "modelImp = $modelImp")
        Log.d("elgin", "connImp = $connImp")
        Log.d("elgin", "paramImp = $paramImp")

        return Termica.AbreConexaoImpressora(typeImp, modelImp, connImp, paramImp)
    }

    fun printerStop() {
        Termica.FechaConexaoImpressora()
    }

    fun avancaLinhas(lines: Int): Int {
        return Termica.AvancaPapel(lines)
    }

    fun cutPaper(lines: Int): Int {
        return Termica.Corte(lines)
    }

    fun imprimeTexto(map: Map<String, Any>): Int {
        val text = map["text"] as String
        val align = map["align"] as Int
        val font = map["font"] as Int
        val fontSize = map["fontSize"] as Int
        return Termica.ImpressaoTexto(text, align, font, fontSize)
    }

    fun imprimeBarCode(map: Map<String, Any>): Int {
        val text = map["text"] as String
        val barCodeType = map["barCodeType"] as Int
        val height = map["height"] as Int
        val width = map["width"] as Int
        val textPosition = map["textPosition"] as Int
        val align = map["align"] as Int
        Termica.DefinePosicao(align)
        return Termica.ImpressaoCodigoBarras(barCodeType, text, height, width, textPosition)
    }

    fun imprimeQR_CODE(map: Map<String, Any>): Int {
        val size = map["size"] as Int
        val text = map["text"] as String
        val align = map["align"] as Int
        val correctionLevel = map["correction"] as Int
        Termica.DefinePosicao(align)
        return Termica.ImpressaoQRCode(text, size, correctionLevel)
    }

    fun printRaw(map: Map<String, Any>): Int {
        val dataRaw = map["data"] as ByteArray
        val totalbytes = map["bytes"] as Int
        val leu = RetornoPOS(1)
        val ler = ByteArray(1000)
        return Termica.DirectIO(dataRaw, totalbytes, ler, leu)
    }

    fun imprimeImagem(map: Map<String, Any>): Int {
        val pathImage = map["path"] as String
        val isBase64 = map["isBase64"] as Boolean
        val bitmap: Bitmap = if (isBase64) {
            val decodedString = Base64.decode(pathImage, Base64.DEFAULT)
            BitmapFactory.decodeByteArray(decodedString, 0, decodedString.size)
        } else {
            val mSaveBit = File(pathImage)
            val filePath = mSaveBit.path
            BitmapFactory.decodeFile(filePath)
        }
        return Termica.ImprimeBitmap(bitmap)
    }

    fun imprimeXMLNFCe(map: Map<String, Any>): Int {
        val xmlNFCe = map["xmlNFCe"] as String
        println(xmlNFCe)
        val indexcsc = map["indexcsc"] as Int
        val csc = map["csc"] as String
        val param = map["param"] as Int
        return Termica.ImprimeXMLNFCe(xmlNFCe, indexcsc, csc, param)
    }

    fun imprimeXMLSAT(map: Map<String, Any>): Int {
        val xml = map["xmlSAT"] as String
        val param = map["param"] as Int
        return Termica.ImprimeXMLSAT(xml, param)
    }

    fun imprimeCupomTEF(cupomTEF: String): Int {
        return Termica.ImprimeCupomTEF(cupomTEF)
    }

    fun statusGaveta(): Int {
        return Termica.StatusImpressora(1)
    }

    fun statusSensorPapel(): Int {
        return Termica.StatusImpressora(3)
    }

    fun statusEjetor(): Int {
        return Termica.StatusImpressora(4)
    }

    fun abrirGavetaElgin(): Int {
        return Termica.AbreGavetaElgin()
    }

    fun abrirGaveta(map: Map<String, Any>): Int {
        val pin = map["pin"] as Int
        val it = map["it"] as Int
        val dp = map["dp"] as Int
        return Termica.AbreGaveta(pin, it, dp)
    }

    fun sinalSonoro(map: Map<String, Any>): Int {
        val qtdSinal = map["times"] as Int
        val st = map["st"] as Int
        val ft = map["ft"] as Int
        return Termica.SinalSonoro(qtdSinal, st, ft)
    }

    fun versaoImpressora(): String {
        return Termica.GetVersaoDLL()
    }

    fun InicializaImpressora(): Int {
        return Termica.InicializaImpressora()
    }
}
